### PR TITLE
supporting version constrain similar to real pip

### PR DESCRIPTION
**Pull request by kpx-dev**
Original PR: https://github.com/ansible/ansible/pull/9064/files?diff=split

### CODE REVIEW

**Comment by gregdek** [comment]
@kienpham2000 thanks for submitting this pull request. Please accept our apologies for taking so long to review it.

It appears that this PR needs to be rebased due to changes in the code base. If you believe that this PR still applies, please rebase and we will then put it into our review process.

**Comment by kpx-dev** [comment]
@gregdek pulled latest devel into it. Thanks

**Comment by gregdek** [comment]
Thanks @kienpham2000.

@mattupstate please review this PR to make sure it adheres to the following guidelines: http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please give it a +1; at that point, we will add the “P1_reviewed” flag to this PR, which will move it to the top of our queue for potential inclusion.

**Comment by kpx-dev** [comment]
forget it, too much work for me. Thanks.

**Comment by gregdek** [comment]
@kienpham2000 the request was for @mattupstate to do the review, because he's the owner of the code. If your PR is good, there shouldn't be any more work required on your part. If you still want to close it, that's your right, of course.

File: packaging/language/pip.py [hunk 1]
```diff
@@ -147,13 +147,21 @@ def _get_cmd_options(module, cmd):
...
-        resp = name
-    else:
+    resp = name
+
+    if version is not None:
+        # this will support version='>1.0.1'
+        for op in ('>', '>=', '<', '<='):
+            # remove all white space so it can support version=' >1.0.1'
+            version = str(version).strip()
+            if version.startswith(op):
```
Comment (sivel) [inline packaging/language/pip.py:160]: Seems that this `startswith` in combination with the for loop is really only checking whether it starts with `>` or `<` due to the order of operations.  Perhaps we shouldn't even perform this check.  The version specs can be rather complex, and validating them won't be easy without using something like `pkg_resources` for doing so.

I'd vote for refactoring this to be more flexible and smarter.

**Comment by mattupstate** [comment]
Why not just let pip do the work?

``` yaml
pip: name=mypackage>=1.0,<2.0 extra_args=-U
```

**Comment by devmage** [comment]
Is this feature still being considered?

**Pull request closed without merging**